java/system-test/system-test-drivers/src/main/java/sleeper/systemtest/drivers/snapshots/AwsSnapshotsDriver.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2022-2024 Crown Copyright
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package sleeper.systemtest.drivers.snapshots;
+
+import com.amazonaws.services.cloudwatchevents.AmazonCloudWatchEvents;
+import com.amazonaws.services.cloudwatchevents.model.DisableRuleRequest;
+import com.amazonaws.services.cloudwatchevents.model.EnableRuleRequest;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import sleeper.configuration.properties.instance.InstanceProperties;
+import sleeper.systemtest.dsl.instance.SystemTestInstanceContext;
+import sleeper.systemtest.dsl.snapshot.SnapshotsDriver;
+
+import static sleeper.configuration.properties.instance.CdkDefinedInstanceProperty.TRANSACTION_LOG_SNAPSHOT_CREATION_RULE;
+
+public class AwsSnapshotsDriver implements SnapshotsDriver {
+    public static final Logger LOGGER = LoggerFactory.getLogger(AwsSnapshotsDriver.class);
+
+    private final InstanceProperties instanceProperties;
+    private final AmazonCloudWatchEvents cwClient;
+
+    public AwsSnapshotsDriver(SystemTestInstanceContext instance, AmazonCloudWatchEvents cwClient) {
+        this.instanceProperties = instance.getInstanceProperties();
```

Comment:
Because this is reading the instance properties immediately, it's difficult to tell whether this is valid without seeing how this will be used in the DSL. For example, if we created this in the SleeperSystemTest constructor and held it in a field, that would break as the instance isn't connected yet.